samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -69,6 +69,115 @@ def full_address(self) -> str:
+    default_variables = module_configuration.get("variables", {})
+    for variable_name, variable_value in default_variables.items():
+        module_variables[variable_name] = ConstantValue(variable_value.get("default"))
+    module_variables.update(input_variables)
+
+    return module_variables
+
+
+def _build_module_resources_from_configuration(module_configuration: Dict, module: TFModule) -> List[TFResource]:
+    module_resources = []
+
+    config_resources = module_configuration.get("resources", [])
+    for config_resource in config_resources:
+        resource_attributes: Dict[str, Expression] = {}
+
+        expressions = config_resource.get("expressions", {})
+        print(expressions)
+        for expression_name, expression_value in expressions.items():
+            parsed_expression = _build_expression_from_configuration(expression_value)
+            resource_attributes[expression_name] = parsed_expression
+
+        resource_address = config_resource.get("address")
+        resource_type = config_resource.get("type")
+        module_resources.append(TFResource(resource_address, resource_type, module, resource_attributes))
+
+    return module_resources
+
+
+def _build_module_outputs_from_configuration(module_configuration: Dict) -> Dict[str, Expression]:
+    module_outputs = {}
+
+    config_outputs = module_configuration.get("outputs", {})
+    for output_name, output_value in config_outputs.items():
+        expression = output_value.get("expression", {})
+        parsed_expression = _build_expression_from_configuration(expression)
+        module_outputs[output_name] = parsed_expression
+
+    return module_outputs
+
+
+def _build_child_modules_from_configuration(module_configuration: Dict, module: TFModule) -> Dict[str, TFModule]:
+    child_modules = {}
+
+    module_calls = module_configuration.get("module_calls", {})
+    for module_call_name, module_call_value in module_calls.items():
+        module_call_input_variables: Dict[str, Expression] = {}
+
+        expressions = module_call_value.get("expressions", {})
+        for expression_name, expression_value in expressions.items():
+            parsed_expression = _build_expression_from_configuration(expression_value)
+            module_call_input_variables[expression_name] = parsed_expression
+
+        module_call_module_config = module_call_value.get("module", {})
+        module_call_built_module = _build_module(
+            module_call_name, module_call_module_config, module_call_input_variables, module.full_address
+        )
+
+        module_call_built_module.parent_module = module
+        child_modules[module_call_name] = module_call_built_module
+
+    return child_modules
+
+
+def _build_expression_from_configuration(expression_configuration: Dict) -> Expression:
+    constant_value = expression_configuration.get("constant_value")
+    references = expression_configuration.get("references")
+
+    parsed_expression: Expression
+
+    if constant_value:
```

Comment:
let's make is `if constant_value is not None`

Reply:
good catch

Reply:
done